src/commands/createWebApp.ts

```diff
@@ -0,0 +1,64 @@
+
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { SiteConfigResource } from "azure-arm-website/lib/models";
+import { ConfigurationTarget, MessageItem, window, workspace, WorkspaceConfiguration } from 'vscode';
+import { AzureParentTreeItem, IActionContext, parseError, SubscriptionTreeItem, UserCancelledError } from "vscode-azureextensionui";
+import { configurationSettings, extensionPrefix } from "../constants";
+import { SiteTreeItem } from "../explorer/SiteTreeItem";
+import { WebAppTreeItem } from "../explorer/WebAppTreeItem";
+import { ext } from "../extensionVariables";
+import { deploy } from "./deploy";
+
+const yesButton: MessageItem = { title: 'Yes' };
+const noButton: MessageItem = { title: 'No', isCloseAffordance: true };
+
+export async function createWebApp(actionContext: IActionContext, node?: AzureParentTreeItem | undefined): Promise<void> {
+    if (!node) {
+        node = <AzureParentTreeItem>await ext.tree.showTreeItemPicker(SubscriptionTreeItem.contextValue);
+    }
+
+    let createdApp: SiteTreeItem | undefined;
+    try {
+        createdApp = <WebAppTreeItem>await node.createChild(actionContext);
+    } catch (error) {
+        const workspaceConfig: WorkspaceConfiguration = workspace.getConfiguration(extensionPrefix);
+        const advancedCreation: boolean | undefined = workspaceConfig.get(configurationSettings.advancedCreation);
+        if (!parseError(error).isUserCancelledError && !advancedCreation) {
+
+            const message: string = `Modify the setting "${extensionPrefix}.${configurationSettings.advancedCreation}" if you want to change the default values when creating a Web App in Azure.`;
+            const btn: MessageItem = { title: 'Turn on advanced creation' };
+            // tslint:disable-next-line: no-floating-promises
+            ext.ui.showWarningMessage(message, btn).then(async result => {
+                if (result === btn) {
+                    await workspaceConfig.update('advancedCreation', true, ConfigurationTarget.Global);
+                }
+            });
+        }
+        throw error;
+    }
+
+    if (createdApp === undefined) {
+        throw new UserCancelledError();
+    }
+
+    createdApp.root.client.getSiteConfig().then(
+        (createdAppConfig: SiteConfigResource) => {
+            actionContext.properties.linuxFxVersion = createdAppConfig.linuxFxVersion ? createdAppConfig.linuxFxVersion : 'undefined';
```

Comment:
If you want to track `linuxFxVersion`, it should be a part of `createChildImpl` so that it's also tracked if they create a web app in the deploy scenario

Reply:
We already track runtime of a created app so this is probably unnecessary now.